File: data-skill/templates/r/plot_bar.R
#!/usr/bin/env Rscript
# Bar Chart Template
# Purpose: Create bar chart for categorical data

library(tidyverse)

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file <- "plot_data.csv"
category_var <- "category"      # Change to your category variable
value_var <- "value"            # Change to your value variable
# ============================================================================

# Load data
df <- read_csv(data_file)

# ============================================================================
# THEME
# ============================================================================
theme_report <- function() {
  theme_minimal(base_size = 11) +
    theme(
      plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
      axis.title = element_text(size = 11, face = "bold"),
      axis.text = element_text(size = 10, color = "black"),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(color = "gray90", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.5),
      plot.margin = margin(10, 10, 10, 10)
    )
}

# WHO color scheme
who_colors <- list(
  blue = "#4D9DE0",
  navy = "#104E8B",
  gray = "#A8A8A8"
)

# ============================================================================
# CREATE PLOT
# ============================================================================
p <- ggplot(df, aes(x = .data[[category_var]], y = .data[[value_var]])) +
  geom_col(fill = who_colors$blue, width = 0.7) +
  geom_text(aes(label = .data[[value_var]]), vjust = -0.5, size = 4) +
  scale_y_continuous(labels = scales::comma_format()) +
  labs(
    title = "Bar Chart Title",
    x = "Category Label",
    y = "Value Label"
  ) +
  theme_report() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# ============================================================================
# SAVE OUTPUT
# ============================================================================
ggsave(
  filename = "bar_chart.png",
  plot = p,
  width = 8,
  height = 6,
  dpi = 300
)

cat("Plot saved to: bar_chart.png\n")

File: data-skill/templates/python/plot_bar.py
#!/usr/bin/env python3
# Bar Chart Template
# Purpose: Create bar chart for categorical data

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file = "plot_data.csv"
category_var = "category"      # Change to your category variable
value_var = "value"            # Change to your value variable
# ============================================================================

# Load data
df = pd.read_csv(data_file)

# ============================================================================
# THEME SETTINGS
# ============================================================================
sns.set_theme(style="whitegrid")
plt.rcParams['font.size'] = 11
plt.rcParams['axes.labelsize'] = 11
plt.rcParams['axes.titlesize'] = 14
plt.rcParams['axes.titleweight'] = 'bold'

# WHO color scheme
who_colors = {
    'blue': '#4D9DE0',
    'navy': '#104E8B'
}

# ============================================================================
# CREATE PLOT
# ============================================================================
fig, ax = plt.subplots(figsize=(8, 6))

bars = sns.barplot(data=df, x=category_var, y=value_var,
                   color=who_colors['blue'], ax=ax)

# Add value labels on bars
for bar in bars.patches:
    height = bar.get_height()
    ax.text(bar.get_x() + bar.get_width()/2., height,
            f'{int(height)}',
            ha='center', va='bottom', fontsize=10)

ax.set_title("Bar Chart Title")
ax.set_xlabel("Category Label")
ax.set_ylabel("Value Label")
ax.spines['top'].set_visible(True)
ax.spines['right'].set_visible(True)

plt.xticks(rotation=45, ha='right')
plt.tight_layout()

# ============================================================================
# SAVE OUTPUT
# ============================================================================
plt.savefig("bar_chart.png", dpi=300, bbox_inches='tight')
print("Plot saved to: bar_chart.png")
plt.close()
